artiq/test/coredevice/test_embedding.py

```diff
@@ -117,6 +118,92 @@ def test_default_arg(self):
         self.assertEqual(exp.run(), 42)
 
 
+class _RPCOverflowTest(EnvExperiment):
+    def build(self):
+        self.setattr_device("core")
+
+    @kernel
+    def run(self, f, g):
+        f(g())
+
+
+class RPCOverflowTest(ExperimentCase):
+    def assert_roundtrip_value_equal(self, g):
+        exp = self.create(_RPCOverflowTest)
+        exp.run(lambda x: self.assertEqual(g(), x), g)
+
+    def test_i32(self):
+        def g() -> TInt32:
+            return (2 ** 31) - 1
+
+        rng = Random(0xdeadbeef)
+        x = rng.randint(-2 ** 31, (2 ** 31) - 1)
```

Comment:
right, it is so tempting to remove it, but how do I prove the function would be correct between (min int32, max int32)? that's why I chose to use a random value and see if it passes, then we could prove it works on all cases

Reply:
Not much difference between ``x`` and ``prng(seed, x)`` for only one value, except that the former is simpler.
I don't see why the code would fail for some intermediate value anyway, testing at the limits should be enough.